src/backend/opencl/kernel/iops.cl

```diff
@@ -35,8 +43,9 @@ void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 #ifdef MIN_OP
 void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 {
-    if ((*lhs > rhs) ||
-        (*lhs == rhs && *lidx < ridx)) {
+    if (!(IS_NAN(*lhs)) &&
```

Comment:
Switch order of isNan operation. See other comments for info.

Reply:
👍